### PR TITLE
185 python port

### LINKED ISSUE #185: Python PORTS should be read from env variables
The ports at which python services are running should be read from environment variables as well as args.
Order of priority:
- args
- environment variable (if no args)
- default value (3000) (if no other option)

### PR DESCRIPTION
**Pull request by tommasodotNET**
Closes #185 

PORT can be passed as env var for both workbench-service and python assistants

The priority order for PORT selection is:
- args
- env variable
- default value

### CODE REVIEW

File: workbench-service/semantic_workbench_service/start.py [hunk 1]
```diff
@@ -33,7 +34,7 @@ def main():
         help="host IP to run service on",
     )
     parse_args.add_argument(
-        "--port", dest="port", type=int, default=settings.service.port, help="port to run service on"
+        "--port", dest="port", type=int, default=int(os.getenv("PORT") or settings.service.port), help="port to run service on"
```
Comment (markwaddle) [inline workbench-service/semantic_workbench_service/start.py:37]: this change is unnecessary. the `settings.service.port` already reads from the `PORT` env var. :)

Comment (markwaddle) [reply workbench-service/semantic_workbench_service/start.py:37]: can you revert this change?

File: libraries/python/semantic-workbench-assistant/semantic_workbench_assistant/start.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def main():
         logger.info("Enabling auto-reload ...")
 
     settings.host = args.host
-    settings.port = args.port or find_free_port(settings.host)
+    settings.port = args.port or int(os.getenv("PORT") or find_free_port(settings.host))
```
Comment (markwaddle) [inline libraries/python/semantic-workbench-assistant/semantic_workbench_assistant/start.py:95]: the settings.port is already designed to read an env var of `ASSISTANT__PORT` (in upper or lower case). to support `PORT` as well, i recommend updating `settings.py`, line 32

```python
    port: int = Field(validation_alias=AliasChoices("PORT", "ASSISTANT__PORT"), default=0)
```

and reverting the change to this file

**Review by markwaddle — CHANGES_REQUESTED** [review]
thanks @tommasodotNET! to keep the implementation consistent with how we're handling env vars today (using Pydantic BaseSettings classes), i recommend a few changes.

**Comment by tommasodotNET** [comment]
> thanks @tommasodotNET! to keep the implementation consistent with how we're handling env vars today (using Pydantic BaseSettings classes), i recommend a few changes.

Hi @markwaddle, thanks for clarifying. Couldn't find the code in which the env var were used. Closing the PR without merging since changes aren't needed.

**Pull request closed without merging**